### PR TITLE
fixed mypy errors in mypy-pandas.tests.extension.decimal.test_decimal

### PR DESCRIPTION
**Pull request by SaturnFromTitan**
Part of #28926
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/extension/decimal/test_decimal.py [hunk 1]
```diff
@@ -103,7 +105,7 @@ def assert_frame_equal(self, left, right, *args, **kwargs):
         decimals = (left.dtypes == "decimal").index
 
         for col in decimals:
-            self.assert_series_equal(left[col], right[col], *args, **kwargs)
+            BaseDecimal.assert_series_equal(left[col], right[col], *args, **kwargs)
```
Comment (jbrockmendel) [inline pandas/tests/extension/decimal/test_decimal.py]: this looks like it should be a classmethod

Comment (SaturnFromTitan) [reply pandas/tests/extension/decimal/test_decimal.py]: What would be the advantage of using `classmethod` over `staticmethod`? I typically only use `classmethod` if I need multiple ways of instantiation

Comment (jbrockmendel) [reply pandas/tests/extension/decimal/test_decimal.py]: having BaseDecimal hard-coded here looks like a kludge, I'm speculating that what you want here is `cls.assert_series_equal`, kind of halfway inbetween the regular method it is now and the staticmethod in the PR

**Comment by WillAyd** [comment]
lgtm outside of @jbrockmendel comments

**Pull request merged**